provider/ethereum/token/client.go

```diff
@@ -587,12 +635,32 @@ func (c *client) standardizeNonFungibleTokenMetadata(_ context.Context, nonFungi
 	return nil
 }
 
+// buildCacheKey builds cache key for Redis.
+func (c *client) buildCacheKey(chainID uint64, address common.Address, id *big.Int) string {
+	var key string
+
+	// Only support Ethereum Virtual Machine (EVM) chain.
+	key = fmt.Sprintf("tokens:ethereum::%d:%s", chainID, address)
+
+	if id != nil {
+		key = fmt.Sprintf("%s:%s", key, id)
+	}
+
+	return key
+}
+
 // NewClient returns a new client.
-func NewClient(ethereumClient ethereum.Client) Client {
+func NewClient(ethereumClient ethereum.Client, options ...Option) Client {
 	instance := client{
 		ethereumClient: ethereumClient,
```

Comment:
We need to use fx for dep injection here.

Reply:
probably a separate PR.

Reply:
Yes will be a separate fx support pr.